addons/redis/scripts/redis-start.sh

```diff
@@ -140,6 +130,93 @@ init_or_get_primary_node() {
   done
 }
 
+init_or_get_primary_from_redis_sentinel() {
+  # check redis sentinel component env
+  if [ -z "$SENTINEL_COMPONENT_NAME" ]; then
+    # return default primary node if redis sentinel component name is not set
+    echo "SENTINEL_COMPONENT_NAME env is not set, try to use default primary node."
+    get_default_initialize_primary_node
+    return
+  fi
+
+  # parse redis sentinel pod list from $SENTINEL_POD_NAME_LIST env
+  if [ -z "$SENTINEL_POD_NAME_LIST" ]; then
+    echo "Error: Required environment variable SENTINEL_POD_NAME_LIST: $SENTINEL_POD_NAME_LIST is not set."
+    exit 1
+  fi
+
+  # get redis sentinel headless service name from $SENTINEL_HEADLESS_SERVICE_NAME env
+  if [ -z "$SENTINEL_HEADLESS_SERVICE_NAME" ]; then
+    echo "Error: Required environment variable SENTINEL_HEADLESS_SERVICE_NAME: $SENTINEL_HEADLESS_SERVICE_NAME is not set."
+    exit 1
+  fi
+
+  old_ifs="$IFS"
+  IFS=','
+  set -f
+  read -ra sentinel_pod_list <<< "${SENTINEL_POD_NAME_LIST}"
+  set +f
+  IFS="$old_ifs"
+  declare -A master_count_map
+  local default_redis_primary_host=""
+  local default_redis_primary_port=""
+  # TODO: further handling of Sentinel exceptions is required, and the startup of Redis should not strongly depend on the Redis Sentinel component.
+  for sentinel_pod in "${sentinel_pod_list[@]}"; do
+    sentinel_pod_fqdn="$sentinel_pod.$SENTINEL_HEADLESS_SERVICE_NAME"
+    # get redis master node from sentinel
+    # shellcheck disable=SC2207
+    REDIS_SENTINEL_INFO=($(redis-cli -h "$sentinel_pod_fqdn" -p "$SENTINEL_SERVICE_PORT" -a "$SENTINEL_PASSWORD" sentinel get-master-addr-by-name "$KB_CLUSTER_COMP_NAME"))
+    echo "sentinel:$sentinel_pod_fqdn has master info: ${REDIS_SENTINEL_INFO[*]}"
+
+    # check if the sentinel has the master info or master info is empty
+    if [ "${#REDIS_SENTINEL_INFO[@]}" -ne 2 ] || [ -z "${REDIS_SENTINEL_INFO[0]}" ] || [ -z "${REDIS_SENTINEL_INFO[1]}" ] ; then
+      echo "sentinel:$sentinel_pod_fqdn has no master info, skip this sentinel."
+      continue
+    fi
+
+    # increment the count of this master in the map
+    host_port_key="${REDIS_SENTINEL_INFO[0]}:${REDIS_SENTINEL_INFO[1]}"
+    master_count_map[$host_port_key]=$(( ${master_count_map[$host_port_key]} + 1 ))
+
+    # track the primary host and port from the first sentinel
+    if [[ -z "$primary_host" ]] && [[ -z "$primary_port" ]]; then
```

Comment:
do you want to express the following logic ?

if [[ -z "$default_redis_primary_host" ]] && [[ -z "$default_redis_primary_port" ]]; then

Reply:
> do you want to express the following logic ?
> 
> if [[ -z "$default_redis_primary_host" ]] && [[ -z "$default_redis_primary_port" ]]; then

yes,  it has been fixed